lib/api/core/entrypoints/embedded/index.js

```diff
@@ -0,0 +1,420 @@
+            index: conf.index,
+            indexPrefix: conf.indexPrefix || 'proxy',
+            indexSuffixPattern: conf.indexSuffixPattern || 'YYYY.MM',
+            messageType: conf.messageType || 'access',
+            ensureMappingTemplate: conf.ensureMappingTemplate !== false,
+            mappingTemplate: conf.mappingTemplate || 'access.log.mapping.json',
+            flushInterval: conf.flushInterval || 2000,
+            clientOpts: conf.clientOpts || {}
+          })));
+          break;
+        case 'file':
+          transports.push(new (winston.transports.File)(Object.assign(opts, {
+            filename: conf.filename || 'kuzzle.access.log',
+            maxSize: conf.maxSize,
+            maxFiles: conf.maxFiles,
+            eol: conf.eol || '\n',
+            logstash: conf.logstash || false,
+            tailable: conf.tailable,
+            maxRetries: conf.maxRetries || 2,
+            zippedArchive: conf.zippedArchive || false
+          })));
+          break;
+        case 'syslog':
+          transports.push(new WinstonSyslog(Object.assign(opts, {
+            host: conf.host || 'localhost',
+            port: conf.port || 514,
+            protocol: conf.protocol || 'udp4',
+            path: conf.path || '/dev/log',
+            pid: conf.pid || process.pid,
+            facility: conf.facility || 'local0',
+            localhost: conf.localhost || 'localhost',
+            type: conf.type || 'BSD',
+            app_name: conf.app_name || process.title,
+            eol: conf.eol
+          })));
+          break;
+        default:
+          // eslint-disable-next-line no-console
+          console.error(`Failed to initialize logger transport "${conf.transport}": unsupported transport. Skipped.`);
+      }
+    }
+
+    this.logger = new (winston.Logger)({transports});
+  }
+
+  joinChannel (channel, connectionId) {
+    debug('[server] client "%s" joining channel "%s"', connectionId, channel);
+
+    const client = this.clients[connectionId];
+
+    if (!client || !client.protocol) {
+      return;
+    }
+
+    try {
+      this.protocols[client.protocol].joinChannel(channel, connectionId);
+    }
+    catch (e) {
+      this.kuzzle.pluginsManager.trigger('log:error', `[join] protocol ${client && client.protocol} failed: ${e.message}`);
+    }
+  }
+
+  /**
+   * @param {Request} request
+   * @param {object} extra
+   */
+  logAccess (request, extra = null) {
+    const
+      connection = this.clients[request.context.connectionId];
+
+    if (!connection) {
+      return this.kuzzle.pluginsManager.trigger('log:warn', `[access log] No connection retrieved for connection id: ${request.context.connectionId}\n` +
+        'Most likely, the connection was closed before the response was received.');
+    }
+
+    if (this.config.logs.accessLogFormat === 'logstash') {
+      // custom kuzzle logs to be exported to logstash
+      this.logger.info({
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Group all shorthand properties at the beginning of this object declaration [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS3499)